main/factor.R

```diff
@@ -0,0 +1,45 @@
+# redirect output to a file
+sink("factor.out")
+
+library(readr)
+library(nFactors)
+library(DataCombine)
+
+# get data
+dat <- read.csv("/dev/stdin")
+write(paste("Number of rows in the data:", nrow(dat)), stdout())
+dat <- DropNA(dat)
+write(paste("Number of rows after droping NAs:", nrow(dat)), stdout())
+write(paste("Number of variables for factor analysis:", ncol(dat)), stdout())
+
+# determine the number of factors
+ev <- eigen(cor(dat, method="spearman", use="complete.obs"))
+write(paste("Eigen values:", ev$values), stdout())
+
+ap <- parallel(subject=nrow(dat), var=ncol(dat), rep=100, cent=0.05)
+nS <- nScree(x=ev$values, aparallel=ap$eigen$qevpea)
+png("scree.png")
+plotnScree(nS)
+
+# try different numbers of factors
+for (n_fac in 3:6){
```

Comment:
what does this range mean?

Reply:
It tries out different numbers of factors - 20 columns down to 3/4/5/6 factors. Then we can look at the output and see which number makes the most sense.